packages/@lwc/template-compiler/src/index.ts

```diff
@@ -23,7 +23,14 @@ import generate from './codegen';
 import { TemplateCompileResult, TemplateParseResult } from './shared/types';
 import { TEMPLATE_MODULES_PARAMETER } from './shared/constants';
 
-export { IRAttributeType, IRElement, TemplateExpression, TemplateIdentifier } from './shared/types';
+export {
```

Comment:
I get the need, but are we okay with/is it safe to expose all of these types externally?

Reply:
I think it is fine to expose those interfaces, for now, to clean up the platform code.

@jmsjtu is working toward making this API public  https://github.com/salesforce/lwc/issues/2432. We will likely change those interfaces in a non-backward compatible fashion soon and that's fine as the `parse` is an internal, non-documented API of the LWC template compiler.